docs/apb-filter-design.md

```diff
@@ -0,0 +1,80 @@
+# APB Filter Design Doc
+
```

Comment:
@jmrodri @rthallisey so what action should I take based on this? Does this document need to get updated, or is it okay to leave it reflective of design thoughts *before* the implementation and update separate official usage documentation for operators?